From c97c96a77b54729f31a5e7b0209d87f314d1d8fe Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Mon, 7 Mar 2016 20:56:07 +0100
Subject: [PATCH] man: ip-neighbour.8: Document all known nud states

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1276661
Upstream Status: iproute2.git commit 948acfed23e5e

commit 948acfed23e5e314fe0f4f863da0368f56645d32
Author: Phil Sutter <phil@nwl.cc>
Date:   Fri Mar 4 20:07:20 2016 +0100

    man: ip-neighbour.8: Document all known nud states

    Not sure how useful they are in practice, but as 'ip neigh' supports
    setting them all, they deserve to be described as well.

    While at it, also add a missing layer of indentation to the subordinate
    nud state list.

    Signed-off-by: Phil Sutter <phil@nwl.cc>
---
 man/man8/ip-neighbour.8 | 22 +++++++++++++++++++++-
 1 file changed, 21 insertions(+), 1 deletion(-)

diff --git a/man/man8/ip-neighbour.8 b/man/man8/ip-neighbour.8
index 19c6d9d..b292e18 100644
--- a/man/man8/ip-neighbour.8
+++ b/man/man8/ip-neighbour.8
@@ -35,7 +35,8 @@ ip-neighbour \- neighbour/arp tables management.
 
 .ti -8
 .IR STATE " := {"
-.BR permanent " | " noarp " | " stale " | " reachable " }"
+.BR permanent " | " noarp " | " stale " | " reachable " | " none " |"
+.BR incomplete " | " delay " | " probe " | " failed " }"
 
 .SH DESCRIPTION
 The
@@ -86,6 +87,7 @@ the state of the neighbour entry.
 is an abbreviation for 'Neighbour Unreachability Detection'.
 The state can take one of the following values:
 
+.RS
 .TP
 .B permanent
 the neighbour entry is valid forever and can be only
@@ -105,6 +107,24 @@ This option to
 .B ip neigh
 does not change the neighbour state if it was valid and the address
 is not changed by this command.
+.TP
+.B none
+this is a pseudo state used when initially creating a neighbour entry or after
+trying to remove it before it becomes free to do so.
+.TP
+.B incomplete
+the neighbour entry has not (yet) been validated/resolved.
+.TP
+.B delay
+neighbor entry validation is currently delayed.
+.TP
+.B probe
+neighbor is being probed.
+.TP
+.B failed
+max number of probes exceeded without success, neighbor validation has
+ultimately failed.
+.RE
 .RE
 
 .TP
-- 
1.8.3.1

